M2/Macaulay2/d/actors5.d

```diff
@@ -1316,9 +1316,19 @@ readlinkfun(e:Expr):Expr := (
      else WrongArgString());
 setupfun("readlink",readlinkfun);
 
+changeDirectory(dir:string):Expr := (
+    if chdir(expandFileName(dir)) == -1
+    then buildErrorPacket(syscallErrorMessage("changing directory"))
+    else nullE);
```

Comment:
You might want to have it return whatever `currentDirectory()` returns. This is useful to know the exact new address.

On that note, it would be good if `makeDirectory()` also returned the address so you could do `changeDirectory makeDirectory "~/foo"`.

Reply:
Great ideas!